backend/LeetWars.Core/LeetWars.Core.BLL/Services/ChallengeService.cs

```diff
@@ -245,6 +245,46 @@ public async Task DeleteChallengeAsync(long challengeId)
             await _context.SaveChangesAsync();
         }
 
+        public async Task SetWeeklyChallenges()
+        {
+            await ResetLastWeeklyChallenge();
+
+            var levels = Enum.GetValues(typeof(LanguageLevel))
+                                            .Cast<LanguageLevel>()
+                                            .ToArray();
+
+            foreach(var level in levels) 
+            {
+                var challengesByLevel = _context.Challenges
+                    .Where(x => x.Level != null && x.Level.SkillLevel == level)
+                    .ToList();
+
+                var randomPosition = GetRandomInt(challengesByLevel.Count);
+                var weeklyChallenge = challengesByLevel.Skip(GetRandomInt(randomPosition)).FirstOrDefault();
```

Comment:
it would be better not to load all challenges but have randomPosition before data loading and just apply it during the challenge selecting

Reply:
our method GetRandomInt is wait maxValue

>  private static int GetRandomInt(int maxValue)

can i use ?
 ```
 foreach(var level in levels) 
  {
      var challengesByLevel = _context.Challenges
          .Where(x => x.Level != null && x.Level.SkillLevel == level)
          .AsQueryable();

      var randomPosition = GetRandomInt(challengesByLevel.Count());
      var weeklyChallenge = challengesByLevel.Skip(randomPosition).FirstOrDefault();
```